c/serverStatusService.c

```diff
@@ -0,0 +1,266 @@
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  struct utsname unameRet;
+  uname(&unameRet);
+  char pid[64];
+  char ppid[64];
+  char dp[64];
+  char cpu_u[64];
+  char mvs_u[64];
+  char ziip_u[64];
+  char zaap_u[64];
+  char* buffer;
+  int rc = 0;
+  int reqtype = 0x00000005; //fullword. request type
+  int rectype = 0x0000004F; //SMF record type, only type 79 is supported
+  int subtype = 0x00000009; //SMF record subtype
+  int bufferlen = 247488; /* length of SMF record buffer */
+  int cpu_util = 0x00000000;
+  int demand_paging = 0x00000000;
+  int options = 0x00000000;
+  int mvs_srm = 0x00000000;
+  int zaap_util = 0x00000000;
+  int ziip_util = 0x00000000;
+  buffer = (char *)safeMalloc(bufferlen, "buffer");
+  memset(buffer, 0, bufferlen);
+  smf_func = (EXSMFI *)fetch("ERBSMFI");
+  rc = (*smf_func)(&reqtype,
+               &rectype,
+               &subtype,
+               buffer,
+               &bufferlen,
+               &cpu_util,
+               &demand_paging,
+               &options,
+               &mvs_srm,
+               &zaap_util,
+               &ziip_util);
+  sprintf(pid, "%d", getpid());
+  sprintf(ppid, "%d", getppid());
+  sprintf(dp, "%d", demand_paging);
+  sprintf(cpu_u, "%d%", cpu_util);
+  sprintf(mvs_u, "%d%", mvs_srm);
+  sprintf(zaap_u, "%d%", zaap_util);
+  sprintf(ziip_u, "%d%", ziip_util);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonAddString(out, "logDirectory", getenv("ZSS_LOG_FILE"));
+  jsonAddString(out, "agentName", "zss");
+  jsonAddString(out, "agentVersion", productVersion);
+  jsonAddString(out, "arch", unameRet.sysname);
+  jsonAddString(out, "osRelease", unameRet.release);
+  jsonAddString(out, "hardwareIdentifier", unameRet.machine);
+  jsonAddString(out, "hostname", unameRet.nodename);
+  jsonStartObject(out, "userEnvironment");
+  char *env_var = strdup(*environ);
+  int i = 1;
+  for (; env_var; i++) {
+    char *var_name = strtok(env_var, "=");
+    char *var_value = strtok(NULL, "=");
+    jsonAddString(out, var_name, var_value);
+    env_var = strdup(*(environ+i));
+  }
+  jsonEndObject(out);
+  jsonAddString(out, "demandPagingRate", dp); 
+  jsonAddString(out, "stdCP_CPU_Util", cpu_u);
+  jsonAddString(out, "stdCP_MVS_SRM_CPU_Util", mvs_u);
+  jsonAddString(out, "ZAAP_CPU_Util", zaap_u);
+  jsonAddString(out, "ZIIP_CPU_Util", ziip_u);
+  jsonAddString(out, "PID", pid);
+  jsonAddString(out, "PPID", ppid);
+  jsonEnd(out);
+  safeFree(env_var, sizeof(*environ)+1);
+  finishResponse(response);
+}
+
+static int serveStatus(HttpService *service, HttpResponse *response) {
+  HttpRequest *request = response->request;
+  JsonObject *dataserviceAuth = jsonObjectGetObject(serverConfig, "dataserviceAuthentication");
+  int rbacParm = jsonObjectGetBoolean(dataserviceAuth, "rbac");
+  if(!rbacParm){
```

Comment:
The name of the variable and the config JSON field is confusing. How is this related to RBAC?

Reply:
The member that states whether RBAC is enabled is under the dataserviceAuthentication member.
```
dataserviceAuthentication{
  "defaultAuth...": "...",
  "rbac": true
  ....
}
```

Reply:
It's just not obvious how this is related to the rest of the code. Why would we not send someone a status response if rbac is disabled in the config?

Reply:
We need to send a response when RBAC is disabled, but it should be an HTTP 400 bad request. This service is conditional on RBAC being enabled, because it is a sensitive URL.

Reply:
That would be a comment I was looking for :-)